iot.py

```diff
@@ -157,56 +197,72 @@ def prompt_for_resource_group(iot):
         sys.exit(1)
 
     iot.set_config("rgroup", name)
-    click.secho("")
 
 def prompt_for_iothub(iot):
     """Prompts the user for an IoT Hub if one isn't passed in"""
     #TODO: deal with clashes both in terms of globally unique names, sku clashes etc
     # right now any error will result in retries
+    name = None
+    click.secho("\nProcessing IoT Hub")
     while True:
-        name = click.prompt("Enter an IoT Hub name")
+        if not name:
+            if not iot.config['iothub']:
+                name = click.prompt("Enter an IoT Hub name")
+            else:
+                name = iot.config['iothub']
+
+        click.secho("Checking for IoT Hub with name '%s'" % name)
         exists, err = run_command_with_stderr_json_out(
             "az iot hub show --resource-group %s --name %s" %
             (iot.config['rgroup'], name))
         if not exists:
-            click.secho("IoTHub with name '%s' does not exist. Creating a new IoTHub..." % name)
+            click.secho("IoT Hub with name '%s' does not exist. Creating a new IoT Hub..." % name)
             output, err = run_command_with_stderr_json_out(
                 "az iot hub create --resource-group %s --name %s --sku %s" %
                 (iot.config['rgroup'], name, iot.config['iothub_sku']))
+            
             if output and "properties" in output and "hostName" in output["properties"]:
                 click.secho("Created a new IoTHub '%s'" % name)
                 hostName = output["properties"]["hostName"]
                 break
             elif 'Bad Request' in err and '400 Client Error' in err and iot.config['iothub_sku'] == 'F1':
-                sku = click.prompt('Unable to use the Free Tier (F1) IoT Hub SKU. '
-                                   'Please choose a different SKU', type=click.Choice(['S1', 'S2', 'S3']))
+                sku = click.prompt('Unable to use the Free Tier (F1) IoT Hub SKU.\n'
+                                   'Please choose a different SKU. e.g. S1, S2, or S3', type=click.Choice(['S1', 'S2', 'S3']))
```

Comment:
Consider adding an explanation on why they're unable to use Free Tier?  e.g. There's only one Free Tier allowed per subscription.